From fd773f7a3c556cb20d6ca8e569f3517905273186 Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Wed, 21 May 2025 08:19:10 +0300
Subject: [PATCH] [JSW-703] jgs: segregate arc specific changes

This patch has made for better code
maintenance. Especially when any function
requires arch specific headers and mem io
calls, it should be in arch specific folder.

Change-Id: Ia8df927beb07d77a159909e69a4e7b9a7da32fdd
---

diff --git a/arch/fs1/arc/dev_spi_tpm.c b/arch/fs1/arc/dev_spi_tpm.c
index 1fd3615..17ebbbb 100644
--- a/arch/fs1/arc/dev_spi_tpm.c
+++ b/arch/fs1/arc/dev_spi_tpm.c
@@ -27,26 +27,6 @@
 	regmap_writel(reg, I0_H0_PSOC_GPIO1(mmGPIO_DATAOUT));
 }
 
-uint32_t spi_tpm_miso_pad_sel_get(void)
-{
-	return IO0_H0_PSOC_PAD_CONF(mmPSOC_PAD_CONF_PAD_SEL_15);
-}
-
-uint32_t spi_tpm_mosi_pad_sel_get(void)
-{
-	return IO0_H0_PSOC_PAD_CONF(mmPSOC_PAD_CONF_PAD_SEL_16);
-}
-
-uint32_t spi_tpm_sclk_out_pad_sel_get(void)
-{
-	return IO0_H0_PSOC_PAD_CONF(mmPSOC_PAD_CONF_PAD_SEL_17);
-}
-
-uint32_t spi_tpm_cs_pad_sel_get(void)
-{
-	return IO0_H0_PSOC_PAD_CONF(mmPSOC_PAD_CONF_PAD_SEL_18);
-}
-
 void spi_tpm_ssi_cs1_get(uint32_t *mask, uint32_t *data_offset,
 				uint32_t *out_en_offset)
 {
@@ -59,3 +39,24 @@
 {
 	return IO0_H0_PSOC_SPI(mmSPI_CTRLR0);
 }
+
+void spi_ssi_cs_init(void)
+{
+	unsigned int reg, mask, data, out_en;
+
+	/* GPIO AF initialization */
+	regmap_writel(1, IO0_H0_PSOC_PAD_CONF(mmPSOC_PAD_CONF_PAD_SEL_15));
+	regmap_writel(1, IO0_H0_PSOC_PAD_CONF(mmPSOC_PAD_CONF_PAD_SEL_16));
+	regmap_writel(1, IO0_H0_PSOC_PAD_CONF(mmPSOC_PAD_CONF_PAD_SEL_17));
+	regmap_writel(0, IO0_H0_PSOC_PAD_CONF(mmPSOC_PAD_CONF_PAD_SEL_18));
+
+	spi_tpm_ssi_cs1_get(&mask, &data, &out_en);
+
+	reg = regmap_readl(data);
+	reg |= mask;
+	regmap_writel(reg, data);
+
+	reg = regmap_readl(out_en);
+	reg |= mask;
+	regmap_writel(reg, out_en);
+}
diff --git a/arch/fs1/arc/dev_spi_tpm.h b/arch/fs1/arc/dev_spi_tpm.h
index fa4bd67..ed81416 100644
--- a/arch/fs1/arc/dev_spi_tpm.h
+++ b/arch/fs1/arc/dev_spi_tpm.h
@@ -17,5 +17,6 @@
 void spi_tpm_ssi_cs1_get(uint32_t *mask, uint32_t *data_offset,
 				uint32_t *out_en_offset);
 uint32_t spi_tpm_base_get(void);
+void spi_ssi_cs_init(void);
 
 #endif /* _DEV_SPI_TPM_H_ */
diff --git a/arch/jgs/arc/dev_spi_tpm.c b/arch/jgs/arc/dev_spi_tpm.c
index db80bbe..0b8ccb4 100644
--- a/arch/jgs/arc/dev_spi_tpm.c
+++ b/arch/jgs/arc/dev_spi_tpm.c
@@ -27,26 +27,6 @@
 	regmap_writel(reg, I0_H0_PSOC_GPIO1(mmGPIO_DATAOUT));
 }
 
-uint32_t spi_tpm_miso_pad_sel_get(void)
-{
-	return IO0_H0_PSOC_PAD_CONF(mmPSOC_PAD_CONF_PAD_SEL_15);
-}
-
-uint32_t spi_tpm_mosi_pad_sel_get(void)
-{
-	return IO0_H0_PSOC_PAD_CONF(mmPSOC_PAD_CONF_PAD_SEL_16);
-}
-
-uint32_t spi_tpm_sclk_out_pad_sel_get(void)
-{
-	return IO0_H0_PSOC_PAD_CONF(mmPSOC_PAD_CONF_PAD_SEL_17);
-}
-
-uint32_t spi_tpm_cs_pad_sel_get(void)
-{
-	return IO0_H0_PSOC_PAD_CONF(mmPSOC_PAD_CONF_PAD_SEL_18);
-}
-
 void spi_tpm_ssi_cs1_get(uint32_t *mask, uint32_t *data_offset,
 				uint32_t *out_en_offset)
 {
@@ -59,3 +39,24 @@
 {
 	return IO0_H0_PSOC_SPI(mmSPI_CTRLR0);
 }
+
+void spi_ssi_cs_init(void)
+{
+	unsigned int reg, mask, data, out_en;
+
+	/* GPIO AF initialization */
+	regmap_writel(1, IO0_H0_PSOC_PAD_CONF(mmPSOC_PAD_CONF_PAD_SEL_15));
+	regmap_writel(1, IO0_H0_PSOC_PAD_CONF(mmPSOC_PAD_CONF_PAD_SEL_16));
+	regmap_writel(1, IO0_H0_PSOC_PAD_CONF(mmPSOC_PAD_CONF_PAD_SEL_17));
+	regmap_writel(0, IO0_H0_PSOC_PAD_CONF(mmPSOC_PAD_CONF_PAD_SEL_18));
+
+	spi_tpm_ssi_cs1_get(&mask, &data, &out_en);
+
+	reg = regmap_readl(data);
+	reg |= mask;
+	regmap_writel(reg, data);
+
+	reg = regmap_readl(out_en);
+	reg |= mask;
+	regmap_writel(reg, out_en);
+}
diff --git a/arch/jgs/arc/dev_spi_tpm.h b/arch/jgs/arc/dev_spi_tpm.h
index fa4bd67..ed81416 100644
--- a/arch/jgs/arc/dev_spi_tpm.h
+++ b/arch/jgs/arc/dev_spi_tpm.h
@@ -17,5 +17,6 @@
 void spi_tpm_ssi_cs1_get(uint32_t *mask, uint32_t *data_offset,
 				uint32_t *out_en_offset);
 uint32_t spi_tpm_base_get(void);
+void spi_ssi_cs_init(void);
 
 #endif /* _DEV_SPI_TPM_H_ */
diff --git a/common/recovery.c b/common/recovery.c
index 11acb2c..842abac 100644
--- a/common/recovery.c
+++ b/common/recovery.c
@@ -9,8 +9,6 @@
 #include <boot_rom.h>
 #include <boot_flow.h>
 #include <common.h>
-#include <global_conf_masks.h>
-#include <global_conf_regs.h>
 #include <pcie.h>
 #include <common.h>
 #include <dev_security.h>
diff --git a/common/tpm2.c b/common/tpm2.c
index 4ba207b..1ba28dd 100644
--- a/common/tpm2.c
+++ b/common/tpm2.c
@@ -5,7 +5,6 @@
  */
 
 #include <boot_rom.h>
-#include <mem_io.h>
 #include <tpm2.h>
 #include <string.h>
 #include <delay.h>
@@ -131,27 +130,6 @@
 static bool tpm_is_supported(void)
 {
 	return true;
-}
-
-static void spi_ssi_cs_init(void)
-{
-	unsigned int reg, mask, data, out_en;
-
-	/* GPIO AF initialization */
-	regmap_writel(1, spi_tpm_miso_pad_sel_get());
-	regmap_writel(1, spi_tpm_mosi_pad_sel_get());
-	regmap_writel(1, spi_tpm_sclk_out_pad_sel_get());
-	regmap_writel(0, spi_tpm_cs_pad_sel_get());
-
-	spi_tpm_ssi_cs1_get(&mask, &data, &out_en);
-
-	reg = regmap_readl(data);
-	reg |= mask;
-	regmap_writel(reg, data);
-
-	reg = regmap_readl(out_en);
-	reg |= mask;
-	regmap_writel(reg, out_en);
 }
 
 static void boot_rom_tpm_spi_local_ctx_init(void)
diff --git a/include/boot_rom.h b/include/boot_rom.h
index 3dbf4f3..13111fd 100644
--- a/include/boot_rom.h
+++ b/include/boot_rom.h
@@ -27,6 +27,8 @@
 #define __unused	__attribute__((__unused__))
 #endif
 
+#define BIT(nr)					(1 << (nr))
+
 #define ARRAY_SIZE(x)				(sizeof(x) / sizeof((x)[0]))
 /* extracts "n" number of bits from position "p" in the given number "v" */
 #define EXTRACT_BITS(v, p, n)	(((v) & (((1 << (n)) - 1) << (p))) >> (p))
diff --git a/include/mem_io.h b/include/mem_io.h
index d5e9bd5..81fef8d 100644
--- a/include/mem_io.h
+++ b/include/mem_io.h
@@ -20,7 +20,6 @@
 
 #define ARC_REG_UPD(OFFSET, ARC_REG_BASE)	((OFFSET & ~(ARC_REG_BASE_MASK)) | ARC_REG_BASE)
 
-#define BIT(nr)					(1 << (nr))
 #define MASK_32_BIT				0xFFFFFFFF
 
 #define DIE_BASED_ADDRESS(offset, die_id)	(offset & ~(HL_ARC_ADDR_DIE_ID_MASK)) | \
